src/source/CppGenerator.scala

```diff
@@ -37,12 +37,12 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
     var hppFwds = mutable.TreeSet[String]()
     var cpp = mutable.TreeSet[String]()
 
-    def find(ty: TypeRef) { find(ty.resolved) }
-    def find(tm: MExpr) {
-      tm.args.foreach(find)
-      find(tm.base)
+    def find(ty: TypeRef, fromInterface: Boolean) { find(ty.resolved, fromInterface) }
```

Comment:
I'd suggest turning this definition around and having the boolean specify what it does, not what it's used for.  I.e. I'd call this "declOnly", or reverse it and call it "needFullDefinition".